src/wallet/wallet.h

```diff
@@ -700,7 +700,16 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
 
     bool fFileBacked;
 
-    std::set<int64_t> setKeyPool;
```

Comment:
I think it would be more straightforward to replace setKeyPool with a `std::map<int64_t, CKeyID>` instead of taking this approach which basically emulates a map using set entries with a custom `<` operator.